.github/workflows/main.yml

```
@@ -32,3 +31,26 @@ jobs:
 
       - name: Check Types
         run: yarn lint:types
+
+  tests:
+    name: Tests
+    runs-on: ubuntu-latest
+
+    strategy:
+        matrix:
+          node-version: [12.x, 14.x, 16.x]
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v3
+
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v3
+      with:
+        cache: 'yarn'
```

i'm not sure how this can implicitly use the matrix's node version - i think you need this:
```suggestion
    - name: Use Node.js ${{ matrix.node-version }}
      uses: actions/setup-node@v3
      with:
        node-version: ${{ matrix.node-version }}
        cache: 'yarn'
```

---

Ah yup, good catch.. missed it in a cleanup